activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -41,18 +41,31 @@ def initialize(scope)
       #
       #    User.where.not(name: %w(Ko1 Nobu))
       #    # SELECT * FROM users WHERE name NOT IN ('Ko1', 'Nobu')
-      #
-      #    User.where.not(name: "Jon", role: "admin")
-      #    # SELECT * FROM users WHERE name != 'Jon' AND role != 'admin'
       def not(opts, *rest)
         opts = sanitize_forbidden_attributes(opts)
 
         where_clause = @scope.send(:where_clause_factory).build(opts, rest)
 
         @scope.references!(PredicateBuilder.references(opts)) if Hash === opts
-        @scope.where_clause += where_clause.invert
+
+        if not_behaves_as_nor?(opts)
+          ActiveSupport::Deprecation.warn(<<~MSG.squish)
+            NOT conditions will no longer behave as NOR in Rails 6.1.
+            To continue using NOR conditions, NOT each conditions manually
```

Comment:
This message is confusing. How about?

```
To continue using NOR conditions, chain each `NOT` condition manually.
```

Reply:
I'm not sure that people want to hurry to migrate to new method, since we have left this issue alone for 5 years (oldest one #14161 I can find).

This issue has already mitigated by 213796f (e9ba12f), and will eventually be fixed even if new method isn't introduced.